client/src/components/Kanban/CardForm/CardForm.tsx

```diff
@@ -0,0 +1,136 @@
+    };
+  }, []);
+
+  return (
+    <Box>
+      {formIsOpen ? <InnerForm columnId={columnId} formAction={setFormIsOpen} /> : null}
+      {!formIsOpen && (
+        <Button
+          onClick={openForm}
+          size="large"
+          color="secondary"
+          disableElevation
+          className={`${!formIsOpen ? classes.button : ''}`}
+        >
+          Add a card...
+        </Button>
+      )}
+    </Box>
+  );
+};
+
+type InnerFormProps = {
+  columnId: string;
+  formAction: Dispatch<SetStateAction<boolean>>;
+};
+const InnerForm = ({ columnId, formAction }: InnerFormProps) => {
+  const [name, setName] = useState<string>('');
+  const [selectedTagColor, setTagColor] = useState<string>('white');
+  const { addCard } = useKanban();
+  const theme = useTheme();
+  const classes = useStyles();
+  return (
+    <>
+      <Box className={classes.cardFormWrapper}>
+        <Box className={classes.cardForm}>
+          <InputBase
+            className={classes.input}
+            value={name}
+            onChange={(e) => setName(e.target.value)}
+            placeholder="Add title..."
+          />
+        </Box>
+        <Divider light />
+        <Box className={classes.tagWrapper}>
+          <Grid container alignItems="center" justify="space-between">
+            <Grid item>
+              <Typography className={classes.typography} variant="body1">
+                Select tag:
+              </Typography>
+            </Grid>
+            <Grid item>
+              <Grid container className={classes.colorsWrapper}>
+                {Object.keys(theme.palette.tags).map((tagColor: string): JSX.Element => {
+                  return (
+                    <Color
+                      key={tagColor}
+                      name={tagColor}
+                      activeSelected={tagColor === selectedTagColor}
+                      setSelected={setTagColor}
+                    />
+                  );
+                })}
+              </Grid>
+            </Grid>
+          </Grid>
+        </Box>
+      </Box>
+      <Button
+        onClick={() => {
+          addCard({
+            name,
+            columnId: columnId,
+            id: `card-${Math.floor(Math.random() * 999999)}`,
+            tag: selectedTagColor,
+          });
+          formAction(false);
+        }}
+        variant="contained"
+        size="large"
+        color="primary"
+        disableElevation
+      >
+        Add a card
+      </Button>
+    </>
+  );
+};
+
+type ColorProps = {
+  name: string;
+  activeSelected: boolean;
+  setSelected: (name: string) => void;
+  key?: string | number;
+};
+
+const Color = ({ name = 'white', activeSelected, setSelected }: ColorProps): JSX.Element => {
+  const classes = useStyles();
+  const colorClasses = useColorTagStyles({ tag: name });
+  return (
+    <Grid
+      onClick={() => setSelected(name)}
+      className={`${classes.color} ${colorClasses.cardTagColor} ${activeSelected && classes.colorSelected}`}
```

Comment:
there's a package for this usecase already included!
https://www.npmjs.com/package/clsx